doc/guides/dts/intro.rst

```diff
@@ -0,0 +1,327 @@
+refer to the labeled node elsewhere in the devicetree. Above, ``a-sub-node``
+has label ``subnode_label``.
+
+Devicetree nodes have *paths* identifying their locations in the tree. Like
+Unix file system paths, devicetree paths are strings separated by slashes
+(``/``), and the root node's path is a single slash: ``/``. Otherwise, each
+node's path is formed by concatenating the node's ancestors' names with the
+node's own name, separated by slashes. For example, the full path to
+``a-sub-node`` is ``/a-node/a-sub-node``.
+
+Devicetree nodes can also have *properties*. Properties are name/value
+pairs. The values are simple byte arrays. Node ``a-sub-node`` has a property
+named ``foo``, whose value is a 32-bit big-endian unsigned integer with value
+3. The size and type of ``foo``\ 's value are implied by the enclosing angle
+brackets (``<`` and ``>``) in the DTS. Refer to the Devicetree Specification
+for a complete list of ways to write a property value in a DTS file.
+
+In practice, devicetree nodes correspond to some hardware, and the node
+hierarchy reflects the hardware's physical layout. For example, let's consider
+a board with three I2C peripherals connected to an I2C bus master on an SoC,
+like this:
+
+.. figure:: zephyr_dt_i2c_high_level.png
+   :alt: representation of a board with three I2C peripherals
+   :figclass: align-center
+
+Nodes corresponding to the I2C bus master and each I2C peripheral would be
+present in this board's devicetree. Reflecting the hardware layout, the
+devicetree's peripheral nodes would be children of the bus master node. Similar
+conventions exist for representing other types of hardware in devicetree.
+
+The corresponding DTS would look something like this:
+
+.. code-block:: none
+
+   / {
+           soc {
+                   i2c-bus-master {
+                           i2c-peripheral-1 {
+                           };
+                           i2c-peripheral-2 {
+                           };
+                           i2c-peripheral-3 {
+                           };
+                   };
+           };
+   };
+
+Properties are used in practice to describe or configure the hardware the node
+represents. For example, an I2C peripheral's node has a property whose value is
+the peripheral's address on the bus.
+
+Here's a tree representing the same example, but with real-world node
+names and properties you might see when working with I2C devices.
+
+.. figure:: zephyr_dt_i2c_example.png
+   :figclass: align-center
+
+   I2C devicetree example with real-world names and properties
+
+Above, node names -- like ``i2c@40003000`` -- are at the top of each node, with
+a gray background, except for the root node, which is shown using its path
+``/``. Properties are shown as ``name=value`` pairs below the node names.
+
+Some important properties are:
+
+compatible
+    Says what kind of device the node represents. The value is a
+    string in the format "vendor,device", like ``"avago,apds9960"``, or a
```

Comment:
`The value is a string in the format "vendor,device"` is not quite correct, we actually also have format "device-type" like: compatible: "gpio-keys", compatible: "gpio-leds", compatible = "mmio-sram", compatible = "fixed-clock".

By the way, I would see it as a big step forward if we could use format "vendor, device, device-type", for example "ti,hdc,sensor-temperature", together with the support for generating static initializers for nodes (#21393).

Reply:
This is pre-existing content not added by this PR, so I will fix by adding a patch.

Reply:
Just some thoughts, I am not sure about this `generic capability` (found it [here](https://github.com/zephyrproject-rtos/zephyr/issues/22964#issuecomment-589161529)), _the recommended format is "manufacturer,model"_.

Reply:
Replaced with:

```
    Says what kind of device the node represents. The recommended format is
    ``"manufacturer,device"``, like ``"avago,apds9960"``, or a sequence of
    these, like ``"ti,hdc", "ti,hdc1010"``. The file
    :zephyr_file:`dts/bindings/vendor-prefixes.txt` contains a list of accepted
    ``manufacturer`` prefixes.
```